pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/index/mutable/MutableTextIndex.java

```diff
@@ -31,12 +31,18 @@ default void add(@Nonnull Object value, int dictId, int docId) {
 
   @Override
   default void add(@Nonnull Object[] values, @Nullable int[] dictIds, int docId) {
-    throw new UnsupportedOperationException("Mutable text indexes are not supported for multi-valued columns");
+    add((String[]) values);
   }
 
   /**
    * Index the document
    * @param document the document as a string
    */
   void add(String document);
+
+  /**
+   * Index the multi-value document
+   * @param document the document as an array of Strings
+   */
+  void add(String[] document);
```

Comment:
(minor) Same for other signatures
```suggestion
  void add(String[] documents);
```

Reply:
thanks for the comments! addressed